ui/ui-components/pages/model/[modelId]/group/[groupId]/rules.tsx

```diff
@@ -231,23 +275,19 @@ export default function Page(props) {
                           as="select"
                           value={rule.key}
                           disabled={loading}
-                          onChange={(e: any) => {
-                            const _rules = [...localRules];
-                            rule.key = e.target.value;
-                            rule.operation.op = "exists"; // every type has an existence check
-                            rule.match = undefined;
-                            rule.relativeMatchNumber = undefined;
-                            rule.relativeMatchUnit = undefined;
-                            rule.relativeMatchDirection = undefined;
-                            rule.topLevel = topLevelGroupRules
-                              .map((tlgr) => tlgr.key)
+                          onChange={(e) => {
+                            const key = e.target.value;
+                            const topLevel = topLevelGroupRules
+                              .map((tlgr) => tlgr.key as string)
```

Comment:
```suggestion
                              .map((tlgr) => tlgr.key)
```
Don't think you need the `as string`

Reply:
I did. because.key is some strange inferred value and it seems that typescript can't do a string comparison